### DISCUSSION TITLE
Incorrect length of GPR file

### DISCUSSION THREAD

**Discussion by Janthony20** [emanuelhuber/RGPR#74] [Bugs] [answered]
Dear RGPR community,
I am having trouble with an impulseradar iprb file. The profile is 242m length. But when I load it into RGPR, it returns a length of 308.74. The window length is correct. Any ideas? I post the output below:
> x<-readGPR(dsn="profile.iprb")
> x

 name        = profile
 filepath    = profile.iprb
 description = 
 survey date = 2022-03-28
 reflection, 70 MHz, Window length = 837.5 ns, dz = 1.25 ns
 30875 traces, 308.74 m

**Comment by emanuelhuber** [comment]
Hello @Janthony20 

Thank you for posting this issue. It is difficult to understand what is going wrong without having a look at your file(s). If you want you can send it to me (emanuel.huber@pm.me).

Do you have a "profile.cor" file where the GPS coordinates are stored in? RGPR will try to estimate the profile length based on this file. If there is no .cor file, RGPR will try to extract the value for the ""USER DISTANCE INTERVAL"-tag in the header file ("profile.iprh"). 
It looks like RGPR found that the trace distance (dx) is equal to 0.01 m. Is it correct?

The profile.iprh and profile.cor are text-files that you can open with a standart text-editor.

Best,

Emanuel

**Reply by Janthony20** [reply]
Hi @emanuelhuber 

Many thanks for your reply. I have sent you an email with the files, including the .iprh and .cor files. Even with these files the software still seems to estimate the length incorrectly.

Kind regards,
Josh

**Comment by emanuelhuber** [comment, marked as answer]
Hello @Janthony20 

The issue you raised (incorrect length of the GPR data) is in fact due to the irregular path of the GPR line: you do not have a straight line but a series of small zig-zags that increase the length of the GPR line (though the distance between the first and the last coordinates is about 237 m (that is similar to the coastline paradox: https://en.wikipedia.org/wiki/Coastline_paradox).

Best,

Emanuel
